### PR TITLE
add basic preprocessor

### PR DESCRIPTION
**Pull request by faneshion**

### CODE REVIEW

**Comment by faneshion** [comment]
This is a basic preprocessor. It can be used by a number of models, such as `ARCI`, `ARCII`, 'MVLSTM`, `DRMMTKS`, `DUET`, `KNRM`, `CONVKNRM`, `Match-LSTM`, and `BiMPM`.

**Comment by faneshion** [comment]
This is a basic preprocessor. It can be used by a number of models, such as `ARCI`, `ARCII`, `MVLSTM`, `DRMMTKS`, `DUET`, `KNRM`, `CONVKNRM`, `Match-LSTM`, and `BiMPM`.

File: matchzoo/preprocessors/basic_preprocessor.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    def __init__(self, fixed_length_left: int = 30,
+                 fixed_length_right: int = 30):
+        """Initialization."""
+        super().__init__()
+        self._fixed_length_left = fixed_length_left
+        self._fixed_length_right = fixed_length_right
+        self._left_fixedlength_unit = processor_units.FixedLengthUnit(
+            self._fixed_length_left, pad_mode='post')
+        self._right_fixedlength_unit = processor_units.FixedLengthUnit(
+            self._fixed_length_right, pad_mode='post')
```
Comment (bwanglzu) [inline matchzoo/preprocessors/basic_preprocessor.py]: can we set `pad_mode` as a parameter?

**Comment by uduse** [comment]
@bwanglzu I think "basic" is totally fine. `BasePreprocessor` is not exposed to user anyways. If any renaming happens, I would rather see changing `BasePreprocessor` to `Preprocessor`.

**Comment by faneshion** [comment]
@bwanglzu `Basic` is indeed a little confusing as there is still `Base`.  However, I think `basic` is more accurate here.

File: matchzoo/preprocessors/basic_preprocessor.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    def fit(self, data_pack: DataPack, verbose=1):
+        """
+        Fit pre-processing context for transformation.
+
+        :param data_pack: data_pack to be preprocessed.
+        :param verbose: Verbosity.
+        :return: class:`DRMMTKSPreprocessor` instance.
```
Comment (yangliuy) [inline matchzoo/preprocessors/basic_preprocessor.py]: Why is the return type `DRMMTKSPreprocessor` insteads of the type for the basic preprocessor ?

File: matchzoo/preprocessors/basic_preprocessor.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        """
+        units = self._default_processor_units()
+        data_pack = data_pack.apply_on_text(chain_transform(units))
+
+        filter_unit = processor_units.FrequencyFilterUnit(low=2, mode='df')
```
Comment (yangliuy) [inline matchzoo/preprocessors/basic_preprocessor.py]: The parameter `low=2` should be configurable from the hyper-parameters.

File: matchzoo/preprocessors/basic_preprocessor.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                                mode='left', inplace=True, verbose=verbose)
+        data_pack.apply_on_text(self._right_fixedlength_unit.transform,
+                                mode='right', inplace=True, verbose=verbose)
+        return data_pack
+
+    @classmethod
+    def _default_processor_units(cls) -> list:
+        """Prepare needed process units."""
+        return [
+            processor_units.TokenizeUnit(),
```
Comment (yangliuy) [inline matchzoo/preprocessors/basic_preprocessor.py]: How about removing stop words ?

Comment (faneshion) [reply matchzoo/preprocessors/basic_preprocessor.py]: Fixed~

**Pull request merged**